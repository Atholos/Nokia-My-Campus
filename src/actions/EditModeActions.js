//used to enable and disable editmode
export const editmode = (value) => {
    return {
        type: 'EDITMODE',
        value
    }
};