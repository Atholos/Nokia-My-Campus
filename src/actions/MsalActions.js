export const msalActions = () => {
    return {
        type: 'SIGNIN',
        identity
    }
};