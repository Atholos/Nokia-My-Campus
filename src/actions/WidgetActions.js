//used to add widgets, used in WidgetReducer
export const increment = (value) =>  {
    return{
        type: 'INCREMENT',
        value
    }
};

//Still not used for anything
export const decrement = (value) => {
    return {
        type: 'DECREMENT',
        value
    };
};


//updates widget
export const updateWidget = (widget) => {
    return {
        type:'UPDATE',
        widget
    };
};

// Removes all widgets, used in WidgetReducer
export const cleanState = () => {
    return {
        type: 'CLEANSTATE'
    };
};