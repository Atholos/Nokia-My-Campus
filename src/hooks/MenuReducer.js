const MenuReducer = (state = [], action) => {

    switch(action.type){
        case 'INCREMENT':
            if(action.value !== [] & !state.includes(action.value)){
                return state.concat(action.value);
            } else {
                return state;
            };

        case 'DECREMENT':
            return state.splice(state.length-1, 1);

        default:
            return state;
    };
};

export default MenuReducer;