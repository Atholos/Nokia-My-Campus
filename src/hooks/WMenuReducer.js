const WMenuReducer = (state=[], action) => {

    if(state === []){
        return state.concat(action.value)
    }
    else{
        return state
    }
};

export default WMenuReducer