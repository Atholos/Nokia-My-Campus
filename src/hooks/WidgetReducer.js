//REDUCERS
//Widget cases, either add or remove widget
const WidgetReducer = (state = [], action) => {

    switch(action.type){
        case 'INCREMENT':
            if(action.value !== []){
                return state.concat(action.value)
            } else {
                break
            };
        
        case 'DECREMENT':
            return state.splice(state.length-1, 1);
        default:
            return state
    };
};


export default WidgetReducer;