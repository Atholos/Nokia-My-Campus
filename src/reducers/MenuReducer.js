const MenuReducer = (state = {}, action) => {

    switch(action.type){

        case 'MENU':
            if(action.value !== {})
                return state = action.value;
            else
                return state;

        default:
            return state;
    }
};

export default MenuReducer;
