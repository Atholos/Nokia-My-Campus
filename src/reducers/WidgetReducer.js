//REDUCERS
//Widget cases, either add or remove widget
const WidgetReducer = (state = [] , action) => {

    switch(action.type){
        case 'INCREMENT':
            
            if(!state.includes(action.value)){
                return state.concat(action.value); 
            } else {
                return state;
            }
        
        case 'DECREMENT':
            return state.splice(state.length-1, 1);

        case 'CLEANSTATE':
            return state = [];
            
        default:
            return state;
    }
};

export default WidgetReducer;