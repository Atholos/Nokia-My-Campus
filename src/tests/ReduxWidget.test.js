import React from 'react';
import * as actions from '../actions/WidgetActions';
import reducer from '../reducers/WidgetReducer';
import { createMount,createShallow } from '@material-ui/core/test-utils';
import Home from '../views/home';

//Needed imports for react redux testing
import { Provider } from 'react-redux'
import UpdateReducer from '../reducers/UpdateReducer'
import WidgetReducer from '../reducers/WidgetReducer';
import WMenuReducer from "../reducers/WMenuReducer";
import MenuReducer from "../reducers/MenuReducer";
import EditModeRecuder from "../reducers/EditModeReducer";
import NewsReducer from "../reducers/NewsReducer";
import { createStore, combineReducers } from 'redux';

describe('action tests', () => {
    let mount;
    let shallow;

    //!!!!!!!!!!! TEST STORE FOR NEWS WITH REDUX !!!!!!!!!!!!!!
  const store = createStore(
    combineReducers({
        UpdateReducer,
        WidgetReducer,
        WMenuReducer,
        MenuReducer,
        EditModeRecuder,
        NewsReducer,
    })
  );

     beforeAll(() => {
        mount = createMount();
        shallow = createShallow();
    });

    afterAll(() => {
        mount.cleanUp();
        shallow.cleanUp();
    });

    //Increment has only value set so it can be anything, an object, int, string etc.
    //this test will check if the incremement action will add the value and create a object, same kind as expectedActions
    it('runs increment action from widgets', () => {
        const value = 1
        const expectedAction = {
            type: 'INCREMENT',
            value
        }
        expect(actions.increment(value)).toEqual(expectedAction);
    });

    //When reducer gets some random value, it will return the default value which is []
    it('returns reducers default state', () => {
        expect(reducer(undefined, '')).toEqual([]);
    });

    //reducer needs a (state, value) in this case we give it an empty state (default value) and a object that is a value
    //redux is supposed to get a list of objects with the value which is why we have brackets [] around value
    it('Integration test for widget action and reducer', () => {
        const value = {
            element: 'widget'
        }
        const expectedAction = {
            type: 'INCREMENT',
            value
        }
        expect(reducer([], expectedAction)).toEqual([value]);
    });

    it('End-to-End (E2E) action test', () => {
        const wrapper = mount(<Provider store={store}><Home/></Provider>)
        expect(wrapper).toMatchSnapshot();
    });

    it('Mounting widget', () => {
        
      });
});