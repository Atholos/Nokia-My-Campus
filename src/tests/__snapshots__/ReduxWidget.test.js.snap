// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action tests End-to-End (E2E) action test 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Home>
    <AuthHome>
      <AuthLoading>
        <div
          style={
            Object {
              "alignItems": "center",
              "bottom": 0,
              "display": "flex",
              "height": "100%",
              "justifyContent": "center",
              "left": 0,
              "margin": "auto 0",
              "position": "absolute",
              "right": 0,
              "textAlign": "center",
              "top": 0,
            }
          }
        >
          <WithStyles(ForwardRef(CircularProgress))
            style={Object {}}
          >
            <ForwardRef(CircularProgress)
              classes={
                Object {
                  "circle": "MuiCircularProgress-circle",
                  "circleDeterminate": "MuiCircularProgress-circleDeterminate",
                  "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
                  "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
                  "circleStatic": "MuiCircularProgress-circleStatic",
                  "colorPrimary": "MuiCircularProgress-colorPrimary",
                  "colorSecondary": "MuiCircularProgress-colorSecondary",
                  "determinate": "MuiCircularProgress-determinate",
                  "indeterminate": "MuiCircularProgress-indeterminate",
                  "root": "MuiCircularProgress-root",
                  "static": "MuiCircularProgress-static",
                  "svg": "MuiCircularProgress-svg",
                }
              }
              style={Object {}}
            >
              <div
                className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                role="progressbar"
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </ForwardRef(CircularProgress)>
          </WithStyles(ForwardRef(CircularProgress))>
        </div>
      </AuthLoading>
    </AuthHome>
  </Home>
</Provider>
`;
